middle_end/inline_and_simplify_aux.mli

```diff
@@ -253,6 +253,12 @@ module Env : sig
 
   (** Print a human-readable version of the given environment. *)
   val print : Format.formatter -> t -> unit
+
+  (** Debuginfo to add to call-sites when inlining a function body. *)
```

Comment:
I propose that:
```ocaml
  (** The environment maintains a list of sites that got inlined to produce
      precise location information.

      When inlining a call-site, call this function to concatenate the
      call-site location to the existing list of sites. *)
  val inline_debuginfo : t -> dbg:Debuginfo.t -> t

  (** Appends the locations of inlined call-sites to the [~dbg] argument *)
  val add_inlined_debuginfo : t -> dbg:Debuginfo.t -> Debuginfo.t
```
The use of "call-sites" now refers to the fact that what gets inlined is a call, not that the `Debuginfo.t` are attached to a call-site (which is too specific as you pointed out).